core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -172,16 +171,10 @@ private boolean checkDirectAccess(AlluxioURI uri) {
     if (!getConf().isSet(PropertyKey.USER_FILE_DIRECT_ACCESS)) {
       return false;
     }
-    if (mPathList == null) {
-      mPathList = getConf().getList(PropertyKey.USER_FILE_DIRECT_ACCESS);
+    if (mPathRegex == null) {
+      mPathRegex = getConf().getString(PropertyKey.USER_FILE_DIRECT_ACCESS);
     }
-    return mPathList.stream().anyMatch(x -> {
-      try {
-        return PathUtils.hasPrefix(uri.getPath(), x);
-      } catch (InvalidPathException e) {
-        return false;
-      }
-    });
+    return uri.getPath().matches(mPathRegex);
```

Comment:
use compiled reg for matching for performance, given this can be used frequently